### ISSUE TITLE
Not ported to libint-2.1.0 yet

### ISSUE THREAD

**Issue by mbanck** [ValeevGroup/mpqc#51]
If I compile the check in external/Libint with libint2-2.1.0 beta2, it works fine on a current Debian unstable (gcc-6.2) system. However, if I upgrade to 2.1.0 final, I get:

```
/$ g++ -I/usr/include/eigen3 -I/usr/include/libint2 /tmp/test.cxx -o /tmp/test -lint2
/tmp/test.cxx: In function 'int main(int, char**)':
/tmp/test.cxx:7:3: error: 'GaussianGmEvalScratch' is not a member of 'libint2'
   libint2::GaussianGmEvalScratch<double,-1> scr;
   ^~~~~~~
/tmp/test.cxx:7:34: error: expected primary-expression before 'double'
   libint2::GaussianGmEvalScratch<double,-1> scr;
                                  ^~~~~~
/tmp/test.cxx:8:3: error: 'scr' was not declared in this scope
   scr.init(5);
   ^~~
```

Further, if I remove that part of the check I get a build failure later on:

```
[ 58%] Building CXX object src/lib/chemistry/qc/libint2/CMakeFiles/libint2.dir/libint2.cc.o
cd "/«PKGBUILDDIR»/obj-x86_64-linux-gnu/src/lib/chemistry/qc/libint2" && /usr/bin/c++   -DHAVE_CONFIG_H -DMPQC_R12 -I"/«PKGBUILDDIR»/obj-x86_64-linux-gnu/src/lib" -I"/«PKGBUILDDIR»/src/lib" -I"/«PKGBUILDDIR»/include" -I/usr/lib/openmpi/include/openmpi -I/usr/lib/openmpi/include/openmpi/opal/mca/event/libevent2022/libevent -I/usr/lib/openmpi/include/openmpi/opal/mca/event/libevent2022/libevent/include -I/usr/lib/openmpi/include -I/usr/include/eigen3 -I/usr/include/libint2 -I/usr/include/openbabel-2.0  -g -O2 -std=c++11    -o CMakeFiles/libint2.dir/libint2.cc.o -c "/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/libint2.cc"
In file included from /«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbintlibint2.h:33:0,
                 from /«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/libint2.cc:36:
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:141:12: warning: 'sc::detail::OSAR_CoreInts<(sc::TwoBodyOper::type)4u>' has a field 'sc::detail::OSAR_CoreInts<(sc::TwoBodyOper::type)4u>::<anonymous>' whose type uses the anonymous namespace [-Wsubobject-linkage]
     struct OSAR_CoreInts<TwoBodyOper::r12_0_g12> : OSAR_CoreInts_G12Base {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:141:12: warning: 'sc::detail::OSAR_CoreInts<(sc::TwoBodyOper::type)4u>' has a base 'sc::detail::{anonymous}::OSAR_CoreInts_G12Base' whose type uses the anonymous namespace [-Wsubobject-linkage]
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:158:12: warning: 'sc::detail::OSAR_CoreInts<(sc::TwoBodyOper::type)5u>' has a field 'sc::detail::OSAR_CoreInts<(sc::TwoBodyOper::type)5u>::<anonymous>' whose type uses the anonymous namespace [-Wsubobject-linkage]
     struct OSAR_CoreInts<TwoBodyOper::r12_m1_g12> : OSAR_CoreInts_G12Base {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:158:12: warning: 'sc::detail::OSAR_CoreInts<(sc::TwoBodyOper::type)5u>' has a base 'sc::detail::{anonymous}::OSAR_CoreInts_G12Base' whose type uses the anonymous namespace [-Wsubobject-linkage]
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:176:12: warning: 'sc::detail::OSAR_CoreInts<(sc::TwoBodyOper::type)8u>' has a field 'sc::detail::OSAR_CoreInts<(sc::TwoBodyOper::type)8u>::<anonymous>' whose type uses the anonymous namespace [-Wsubobject-linkage]
     struct OSAR_CoreInts<TwoBodyOper::g12t1g12> : OSAR_CoreInts_G12Base {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:176:12: warning: 'sc::detail::OSAR_CoreInts<(sc::TwoBodyOper::type)8u>' has a base 'sc::detail::{anonymous}::OSAR_CoreInts_G12Base' whose type uses the anonymous namespace [-Wsubobject-linkage]
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:291:14: error: 'GaussianGmEvalScratch' in namespace 'libint2' does not name a template type
     libint2::GaussianGmEvalScratch<double, -1> coreints_scratch_;
              ^~~~~~~~~~~~~~~~~~~~~
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h: In constructor 'sc::TwoBodyOSARLibint2<OperType>::TwoBodyOSARLibint2(sc::Integral*, const sc::Ref<sc::GaussianBasisSet>&, const sc::Ref<sc::GaussianBasisSet>&, const sc::Ref<sc::GaussianBasisSet>&, const sc::Ref<sc::GaussianBasisSet>&, size_t, const sc::Ref<sc::IntParams>&)':
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:411:5: error: 'coreints_scratch_' was not declared in this scope
     coreints_scratch_.init(l1+l2+l3+l4);
     ^~~~~~~~~~~~~~~~~
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h: In copy constructor 'sc::TwoBodyOSARLibint2<OperType>::TwoBodyOSARLibint2(const sc::TwoBodyOSARLibint2<OperType>&)':
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:492:5: error: 'coreints_scratch_' was not declared in this scope
     coreints_scratch_.init(l1+l2+l3+l4);
     ^~~~~~~~~~~~~~~~~
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h: In member function 'size_t sc::TwoBodyOSARLibint2<OperType>::quartet_data_(sc::TwoBodyOSARLibint2<OperType>::prim_data*, double)':
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:632:59: error: 'coreints_scratch_' was not declared in this scope
                                       static_cast<void*>(&coreints_scratch_));
                                                           ^~~~~~~~~~~~~~~~~
/«PKGBUILDDIR»/src/lib/chemistry/qc/libint2/tbosar.h:653:59: error: 'coreints_scratch_' was not declared in this scope
                                       static_cast<void*>(&coreints_scratch_));
                                                           ^~~~~~~~~~~~~~~~~
src/lib/chemistry/qc/libint2/CMakeFiles/libint2.dir/build.make:473: recipe for target 'src/lib/chemistry/qc/libint2/CMakeFiles/libint2.dir/libint2.cc.o' failed
make[3]: *** [src/lib/chemistry/qc/libint2/CMakeFiles/libint2.dir/libint2.cc.o] Error 1
```

It appears mpqc3 does not work with the latest stable version of libint2?

**Comment by mbanck** [comment]
Any update on this?  What is the recommended version of Libint2 to be used with MPQC master?

**Comment by evaleev** [comment]
updated ... 9c3df01f50be9300f0ddd45dedc4b5faf31d7bd6 requires just released libint 2.3.0-beta.3
